### ISSUE TITLE
error loading `@iconify/svelte/dist/offline` in sveltekit

### ISSUE THREAD

**Issue by tomyan** [iconify/iconify#102]
The following works fine in my sveltekit app:

```
import Icon from '@iconify/svelte';
```

However, when I change it to:

```
import Icon from '@iconify/svelte/dist/offline';
```

...then I get this error:

```
Error: <Icon> is not a valid SSR component. You may need to review your build config to ensure that dependencies are compiled, rather than imported as pre-compiled modules
```

**Comment by cyberalien** [comment]
Try this:
```ts
import Icon from '@iconify/svelte/dist/OfflineIcon.svelte';
```

**Comment by tylkomat** [comment]
Since this is still an open issue I just post here. I import my Icon using:
```
<script>
  import Icon from '@iconify/svelte/dist/OfflineIcon.svelte';
  import userIcon from '@iconify-icons/bx/bx-user';
</script>

<Icon icon{userIcon} />
```

This works fine in dev mode.
When I build the app using `adapter-node` and visit the page I get the following error:
```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/app/node_modules/@iconify-icons/bx/bx-user' imported from /app/server/entries/pages/__layout.svelte.js
Did you mean to import @iconify-icons/bx/bx-user.js?
```
When I arrive at the same page using client side navigation then it works fine.

**Comment by cyberalien** [comment]
Try adding `.js` extension to import to point to exact file.

Though it looks like a bug in whatever is used to build files and map imports - it should have resolved import and added `.js` extension.

**Comment by tylkomat** [comment]
Adding `.js` extension has the same result.

**Comment by cyberalien** [comment]
Is it added as dev dependency or production? Since it is used in production, not bundled, make sure it is installed as production dependency.

**Comment by tylkomat** [comment]
TLDR: the packages are in the node_modules folder.

It's in dev dependencies, but I do docker multistage build and copy the node_modules folder into the next stage to have all dependencies there.

**Comment by cyberalien** [comment]
Then I have no more ideas on how to solve this. Structure sounds solid, import should have worked correctly.

Another possible solution is to try `unplugin-icons` instead: https://github.com/antfu/unplugin-icons

It is similar component, uses same icons data from Iconify. Same as offline component, it bundles all icon data.

**Comment by tylkomat** [comment]
Thanks. I'll try.

**Comment by tylkomat** [comment]
I was digging more and you were right. It was indeed the issue of the missing `.js`. I don't know why it didn't do anything yesterday. Thanks for the help.
